pkgs/native_assets_builder/lib/src/build_runner/build_runner.dart

```diff
@@ -437,12 +437,12 @@ class NativeAssetsBuildRunner {
         ),
       );
       if (buildOutput == null) return null;
-      hookResult = hookResult.copyAdd(buildOutput);
+      hookResult = hookResult.copyAdd(buildOutput, [/*dry run is not cached*/]);
     }
     return hookResult;
   }
 
-  Future<HookOutput?> _runHookForPackageCached(
+  Future<(HookOutput?, List<Uri>?)> _runHookForPackageCached(
```

Comment:
This method always returns either `(null, null)` or `(non-null, non-null)`. This complicates call sites that check for `ret.$1 != null` and then using `ret.$2!` (bang operator).

Also all call sites never use `ret.$2` if `ret.$1 == null`.

=> It would simplify things if this would return `(HookOutput, List<Uri>)?` (i.e. a nullable record instead of a non-nullable record with nullable fields)